auth/src/main/java/org/aerogear/auth/impl/UserPrincipalImpl.java

```diff
@@ -137,14 +137,22 @@ public String getName() {
         return username;
     }
 
+    /**
+     * Check whether the user is authorized and not expired.
+     * @return <code>true</code> if user is authorized and token is not expired.
+     */
+    @Override
+    public boolean checkValidAuth() {
```

Comment:
I think it would be better to add this method to the `credentials` object so that the `UserPrincipalImpl` remains general purpose. WDYT?

Reply:
Ah I understand what you mean now, yep that makes sense to me